packages/meditrak-server/src/routes/postChanges.js

```diff
@@ -72,6 +80,18 @@ const PAYLOAD_VALIDATORS = {
   },
 };
 
+/**
+ * Contains functions that accept the payload, and translate the data so that the required fields are present
+ * for the relevant change action, returning true if successful and throwing an error if not
+ */
+const PAYLOAD_TRANSLATORS = {
+  [SUBMIT_SURVEY_RESPONSE]: async (models, payload) =>
```

Comment:
Pretty clean and matches the validator pattern, but I would prefer it wasn't mutating tbh. Do you think it would be a big performance impact to deep copy the payload and return the translated version?

Reply:
Probably wouldn't be the biggest performance hit, though it would affect all survey_responses submitted via this endpoint. Given that almost all of these come from meditrak-app and don't require any translation, doing a deep copy for each seems a bit risky...
What I might do is some kind of upfront check whether or not a response requires any translation before copying, which I think might even be more performant that what we have currently 👍

Reply:
Oh yeah true, it would affect all those meditrak changes. Cool a check sounds good, in the absence of moving to a completely different endpoint handler.